packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewManager.java

```diff
@@ -107,9 +107,9 @@ public void setBackgroundImage(ReactViewGroup view, @Nullable ReadableArray back
           ReadableMap gradientMap = backgroundImage.getMap(i);
           gradients[i] = new Gradient(gradientMap);
         }
-        view.setGradients(gradients);
+        view.setBackgroundImage(gradients);
```

Comment:
The greater context behind BackgroundStyleApplicator is that it works on any view (e.g. Text instead of just View). Once the flag has been enabled a bit longer, my plan has been to start moving all the setters to BaseViewMansger.

This is why the current view managers use BackgroundStyleApplicator directly when the gate is on, instead of calling through any specific underlying views.